browser/extensions/api/brave_rewards_api.cc

```diff
@@ -431,49 +417,95 @@ ExtensionFunction::ResponseAction BraveRewardsGetCurrentReportFunction::Run() {
   return RespondNow(NoArguments());
 }
 
-BraveRewardsGetGrantsFunction::~BraveRewardsGetGrantsFunction() {
+BraveRewardsFetchPromotionsFunction::~BraveRewardsFetchPromotionsFunction() {
 }
 
-ExtensionFunction::ResponseAction BraveRewardsGetGrantsFunction::Run() {
+ExtensionFunction::ResponseAction BraveRewardsFetchPromotionsFunction::Run() {
   Profile* profile = Profile::FromBrowserContext(browser_context());
   RewardsService* rewards_service =
     RewardsServiceFactory::GetForProfile(profile);
   if (rewards_service) {
-    rewards_service->FetchGrants(std::string(), std::string());
+    rewards_service->FetchPromotions();
   }
   return RespondNow(NoArguments());
 }
 
-BraveRewardsGetGrantCaptchaFunction::~BraveRewardsGetGrantCaptchaFunction() {
+BraveRewardsClaimPromotionFunction::~BraveRewardsClaimPromotionFunction() {
 }
 
-ExtensionFunction::ResponseAction BraveRewardsGetGrantCaptchaFunction::Run() {
-  std::unique_ptr<brave_rewards::GetGrantCaptcha::Params> params(
-      brave_rewards::GetGrantCaptcha::Params::Create(*args_));
+ExtensionFunction::ResponseAction BraveRewardsClaimPromotionFunction::Run() {
+  std::unique_ptr<brave_rewards::ClaimPromotion::Params> params(
+      brave_rewards::ClaimPromotion::Params::Create(*args_));
   Profile* profile = Profile::FromBrowserContext(browser_context());
   RewardsService* rewards_service =
     RewardsServiceFactory::GetForProfile(profile);
-  if (rewards_service) {
-    rewards_service->GetGrantCaptcha(params->promotion_id,
-                                     params->type);
+  if (!rewards_service) {
+    auto data = std::make_unique<base::Value>(base::Value::Type::DICTIONARY);
+    data->SetIntKey("result", 1);
+    return RespondNow(OneArgument(std::move(data)));
   }
-  return RespondNow(NoArguments());
+  rewards_service->ClaimPromotion(
+      base::BindOnce(
+        &BraveRewardsClaimPromotionFunction::OnClaimPromotion,
+        this,
+        params->promotion_id));
+  return RespondLater();
+}
+
+void BraveRewardsClaimPromotionFunction::OnClaimPromotion(
+    const std::string& promotion_id,
+    const int32_t result,
+    const std::string& captcha_image,
+    const std::string& hint,
+    const std::string& captcha_id) {
+
+  auto data = std::make_unique<base::Value>(base::Value::Type::DICTIONARY);
+  data->SetIntKey("result", result);
+  data->SetStringKey("promotionId", promotion_id);
+  data->SetStringKey("captchaImage", captcha_image);
+  data->SetStringKey("captchaId", captcha_id);
+  data->SetStringKey("hint", hint);
+  Respond(OneArgument(std::move(data)));
```

Comment:
These are used in multiple locations so would be better as const strings

Reply:
can you please log an issue for it. Would be good to think about this problem in general

Reply:
https://github.com/brave/brave-browser/issues/6950